FlattenException.php

```diff
@@ -0,0 +1,242 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\debug;
+
+use yii\base\Object;
+use yii\web\HttpException;
+
+/**
+ * FlattenException wraps a PHP Exception to be able to serialize it.
+ * Basically, this class removes all objects from the trace.
+ * Ported from Symfony components
+ *
+ * @author Dmitry Bashkarev <dmitry@bashkarev.com>
+ * @since 2.0.10
+ */
+class FlattenException extends Object
+{
+    private $_message;
+    private $_code;
+    private $_previous;
+    private $_trace;
+    private $_class;
+    private $_statusCode;
+    private $_file;
+    private $_line;
+
+    public static function create(\Exception $exception, $statusCode = null)
+    {
+        $e = new static();
+        $e->setMessage($exception->getMessage());
+        $e->setCode($exception->getCode());
+
+        if ($exception instanceof HttpException) {
+            $statusCode = $exception->statusCode;
+        }
+
+        if (null === $statusCode) {
+            $statusCode = 500;
+        }
+
+        $e->setStatusCode($statusCode);
+        $e->setTraceFromException($exception);
+        $e->setClass(get_class($exception));
+        $e->setFile($exception->getFile());
+        $e->setLine($exception->getLine());
+
+        $previous = $exception->getPrevious();
+
+        if ($previous instanceof \Exception) {
+            $e->setPrevious(static::create($previous));
+        }
+        return $e;
+    }
+
+    public function toArray()
+    {
+        $exceptions = [];
+        foreach (array_merge([$this], $this->getAllPrevious()) as $exception) {
+            $exceptions[] = [
+                'message' => $exception->getMessage(),
+                'class' => $exception->getClass(),
+                'trace' => $exception->getTrace(),
+            ];
+        }
+
+        return $exceptions;
+    }
+
+    public function getStatusCode()
+    {
+        return $this->_statusCode;
+    }
+
+    public function setStatusCode($code)
+    {
+        $this->_statusCode = $code;
+    }
+
+    public function getClass()
+    {
+        return $this->_class;
+    }
+
+    public function setClass($class)
```

Comment:
Yii's approach is to just use public variables for variables that are... public.